test/runtests.jl

```diff
@@ -251,6 +251,9 @@ end
 @test sprint(JSON.print, [NaN]) == "[null]"
 @test sprint(JSON.print, [Inf]) == "[null]"
 
+# check for issue #163
+@test isapprox(JSON.parse(json(Float32(2.1e-8))), 2.1e-8)
```

Comment:
That's correct. In fact `2.1f8 != 2.1e8`. I could probably play with numbers to find one that's more amenable to lossless round-tripping, but that doesn't feel any cleaner than using `isapprox`.